app/src/main/kotlin/com/intive/picover/profile/delete/viewmodel/DeleteAccountViewModel.kt

```diff
@@ -0,0 +1,30 @@
+package com.intive.picover.profile.delete.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.intive.picover.R
+import com.intive.picover.auth.model.AccountDeletionResult
+import com.intive.picover.auth.repository.AuthRepository
+import com.intive.picover.common.toast.ToastPublisher
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.launch
+
+@HiltViewModel
+class DeleteAccountViewModel @Inject constructor(
+	private val authRepository: AuthRepository,
+	private val toastPublisher: ToastPublisher,
+) : ViewModel() {
+
+	fun onDeleteAccountClick() {
+		viewModelScope.launch {
+			val accountDeletionResult = authRepository.deleteAccount()
+			when (accountDeletionResult) {
+				is AccountDeletionResult.Success -> R.string.DeleteAccountSuccessToastText
+				is AccountDeletionResult.ReAuthenticationNeeded -> R.string.DeleteAccountReAuthenticationToastText
+			}.let {
+				toastPublisher.show(it)
```

Comment:
It's a good time to get rid of toast publisher at all and use only snack bars, since it's already configured to use

Reply:
ToastPublisher deleted but the migration to snack bar will not be a part of this issue because:
1. It is introducing many changes (SnackBarState needs to be passed to dialog),
2. Snackbar will not be shown in case of logout because we are using Firebase Activity (FirebaseUI).

Reply:
1. and the conclusion of that should be that handling of side effects it's no the best one 
2. I vote for removing this message, it's not needed

Reply:
I think the message is crucial here because in case you need to re-authenticate, the user needs to log in again and then delete the account so I'm voting to keep the toast for now. Later on if we decide to implement our own authentication screen then this feature could be done in other way.

Reply:
Thanks for the explanation, I thought that only "logout" case it's affected. Let's keep it as you suggest